projection/src/test/scala/akka/projection/r2dbc/R2dbcTimestampOffsetProjectionSpec.scala

```diff
@@ -726,39 +723,29 @@ class R2dbcTimestampOffsetProjectionSpec
         }
       }
 
-      val startTime = Instant.now()
-      val envelopes1 = createEnvelopesUnknownSequenceNumbers(startTime, pid1, pid2)
-
       val projection1 =
-        R2dbcProjection.groupedWithinAsync(
-          projectionId,
-          Some(settings),
-          sourceProvider = sourceProvider(envelopes1),
-          handler = () => handler())
+        R2dbcProjection.groupedWithinAsync(projectionId, Some(settings), sourceProvider1, handler = () => handler())
 
       projectionTestKit.run(projection1) {
         result1.toString shouldBe "e1-1|e1-2|e1-3|"
-        result2.toString shouldBe empty
+        result2.toString shouldBe "e2-1|"
       }
 
       // simulate backtracking
       logger.debug("Starting backtracking")
       val envelopes2 = createEnvelopesBacktrackingUnknownSequenceNumbers(startTime, pid1, pid2)
+      val sourceProvider2 = createBacktrackingSourceProvider(envelopes2)
       val projection2 =
-        R2dbcProjection.groupedWithinAsync(
-          projectionId,
-          Some(settings),
-          sourceProvider = backtrackingSourceProvider(envelopes2),
-          handler = () => handler())
+        R2dbcProjection.groupedWithinAsync(projectionId, Some(settings), sourceProvider2, handler = () => handler())
 
       projectionTestKit.run(projection2) {
-        // FIXME result1 is "e1-1|e1-2|e1-3|e1-2|e1-3|e1-4|e1-5|e1-6|
+        // FIXME
```

Comment:
Is this a different problem or would it also be fixed by this PR?

Reply:
I tried to enable it but it's still failing so will investigate separately (it's for grouped async case so not top prio)